src/main/java/com/codeus/winter/config/DefaultBeanFactory.java

```diff
@@ -204,15 +187,17 @@ public void initializeBeans() {
      * @param beanName       a name of a bean to retrieve.
      * @param beanDefinition a definition of a bean to retrieve.
      * @return a bean instance.
-     * @throws IllegalArgumentException if given beanDefinition has {@link Scope#PROTOTYPE} scope.
+     * @throws IllegalArgumentException if given beanDefinition has unsupported scope.
      */
     private Object getBean(String beanName, BeanDefinition beanDefinition) {
         if (beanDefinition.isSingleton()) {
             return getSingleton(beanName, beanDefinition);
+        } else if (beanDefinition.isPrototype()) {
```

Comment:
[minor], here can be used strategy pattern because we can have more than 2 scopes [singleton, prototype, request, application, etc..]

Reply:
I believe it would add unnecessary complexity to the existing code. Therefore it is better to implement strategy when other scopes are added.